app/models/api_key.rb

```diff
@@ -34,7 +42,30 @@ def generate_access_token
     end
   end
 
+  def create_bot_user
+    if self.bot_user.blank? && self.team.present?
+      bot_name = "#{self.team.slug}-bot-#{self.title}"
+      new_bot_user = BotUser.new(api_key: self, name: bot_name, login: bot_name)
+      new_bot_user.skip_check_ability = true
+      new_bot_user.set_role 'editor'
```

Comment:
I was not sure if this would indeed create a `TeamUser` instance but I confirmed it does.